gtsam/linear/NoiseModel.h

```diff
@@ -751,15 +750,8 @@ namespace gtsam {
         typedef boost::shared_ptr<Fair> shared_ptr;
 
         Fair(double c = 1.3998, const ReweightScheme reweight = Block);
-        double weight(double error) const {
-          return 1.0 / (1.0 + std::abs(error) / c_);
-        }
-        double residual(double error) const {
-          double absError = std::abs(error);
-          double normalizedError = absError / c_;
-          double val = normalizedError - std::log(1 + normalizedError);
-          return c_ * c_ * val;
-        }
+        double weight(double error) const;
```

Comment:
Const then? But, I normally consider const redundant on arguments passed by value...

Reply:
Yeah I was wondering what your policy on that was? Some people say const in the declaration is redundant since it doesn't do anything. Should I remove the const for the function arguments then? I felt it was a nice built-in check since the values should not change.

Reply:
I would say undo these changes. We don’t do it anywhere else in GTSAM, and it is orthogonal to the goal of the PR

Reply:
Understood.